tools/ExtensionsMetadataGenerator/test/ExtensionsMetadataGeneratorTests/ExtensionsMetadataGeneratorTests.cs

```diff
@@ -42,17 +46,16 @@ public void Generator_DifferentTargetFrameworks()
                 // plus the two from the 2.1 and 2.2 test projects.
                 JToken extensions = json["extensions"];
                 int startups = extensions.Count();
-                Assert.Equal(4, startups);
+                Assert.Equal(5, startups);
 
                 Assert.Single(extensions, e => e["name"].ToString() == "Foo" && e["typeName"].ToString().StartsWith("ExtensionsMetadataGeneratorTests.FooWebJobsStartup, ExtensionsMetadataGeneratorTests"));
                 Assert.Single(extensions, e => e["name"].ToString() == "BarExtension" && e["typeName"].ToString().StartsWith("ExtensionsMetadataGeneratorTests.BarWebJobsStartup, ExtensionsMetadataGeneratorTests"));
                 Assert.Single(extensions, e => e["name"].ToString() == "Startup" && e["typeName"].ToString().StartsWith("TestProject_Core21.Startup"));
                 Assert.Single(extensions, e => e["name"].ToString() == "Startup" && e["typeName"].ToString().StartsWith("TestProject_Core22.Startup"));
+                Assert.Single(extensions, e => e["name"].ToString() == "Startup" && e["typeName"].ToString().StartsWith("TestProject_Razor.Startup"));
```

Comment:
Note that this scenario does not currently work. Today if we find an assembly-level attribute and we cannot determine it's base type (Razor is the example that I've built; it does not get copied to bin), we would have to completely bail on processing that assembly. With Cecil, we can skip that attribute and actually discover any relevant startup attributes.